packages/flutter_tools/lib/src/test/runner.dart

```diff
@@ -89,14 +123,65 @@ Future<int> runTests(
 
   // Call package:test's main method in the appropriate directory.
   final Directory saved = fs.currentDirectory;
+
+  final Function printReloadMessage = () {
+    printStatus(
+      "Press 'r' to rerun your tests, 'q' to quit",
+      color: TerminalColor.red
+    );
+  };
+
   try {
     if (workDir != null) {
       printTrace('switching to directory $workDir to run tests');
       fs.currentDirectory = workDir;
     }
 
     printTrace('running test package with arguments: $testArgs');
-    await test.main(testArgs);
+    await test.runTests(testArgs);
+
+    if (watchTests) {
+      final Completer<void> completer = Completer<void>();
+      final DirectoryWatcher directoryWatcher = DirectoryWatcher(saved.path);
+
+      directoryWatcher.events.listen((WatchEvent event) {
+        if (event.path.endsWith('.dart')) {
+          invalidatedFiles.add(event.path);
+        }
+      });
+
+      terminal.singleCharMode = true;
+      terminal.keystrokes.asBroadcastStream().listen((String char) async {
+        switch(char) {
+          case 'r':
+            final Status status =
+                logger.startProgress('Recompiling test files...\n', timeout: null);
```

Comment:
I don't think the newline is necessary here

Reply:
Done